### ISSUE TITLE
TIN results are different depending on the extent used for the interpolation

### ISSUE THREAD

**Issue by queenmedley** [qgis/QGIS#37871]
The difference in grid values ​​occurs when the entire shape file range is interpolated and when only the specific part is interpolated.

1. When creating a shapefile, selecting the extent option from "calculator from layer" produce results similar to the actual contour height. 

2. However, when selecting a specific part with the extent option set to "Draw on Canvas", the error occurs somewhat.

![sample](https://user-images.githubusercontent.com/16573046/87863122-a80c2d80-c992-11ea-88eb-a82fbbbdb63a.jpg)

[contour_edit.zip](https://github.com/qgis/QGIS/files/4942821/contour_edit.zip)

* attach file CRS is EPSG:5187

**Comment by gioman** [comment]
@queenmedley what tool? TIN interpolation?

If I have understand correctly:

interpolating using as extent the extent of your layer "contour edit" produces values that more similar to the Z values of input lines, while if is used a smaller extent (i.e. the extent of "plan") results are slightly different, Is that correct?

what are the parameters you are using in the TIN tool?

**Comment by queenmedley** [comment]
@gioman

What you understand is correct.

The method I used is "TIN interpolation".
The "interpolation attribute" of the attached file is "z_first", 
"Linear" is used for the "interpolation method",
and pixel size is 10.

**Comment by gioman** [comment]
@queenmedley QGIS version used? Operatung system?

**Comment by queenmedley** [comment]
@gioman

I use Qgis 3.14 and OS is Windows 10.
I have linux desktop, I will test it on Linux and answer the result.

**Comment by gioman** [comment]
@queenmedley any chance you can test on QGIS master? It is always crashing for me when I try to interpolate your data.

**Comment by queenmedley** [comment]
I'm sorry. I don't know what "master" is.
Previously, the contoure file seems problematic.
So I just ran "interpolation" with 3 different cases and still had a slight difference of around 1m. It is not large compared to the previous 10m.
I don't know if it is normal to make a difference in interpolation even if I set a sufficient range around the target range.

**3cases are pa, mid, full, and min and max value made by "zonal stastic" are different...  
![interpol](https://user-images.githubusercontent.com/16573046/87938641-d93f4780-cad1-11ea-9682-036d658343db.png)
![zonal_stat](https://user-images.githubusercontent.com/16573046/87938656-dd6b6500-cad1-11ea-892a-99f9b71726c5.png)
[interpol.tar.gz](https://github.com/qgis/QGIS/files/4947734/interpol.tar.gz)
